c++/src/kj/common.h

```diff
@@ -373,6 +373,23 @@ struct DisallowConstCopy {
   DisallowConstCopy& operator=(DisallowConstCopy&&) = default;
 };
 
+#if _MSC_VER
+
+#define MSVC_CAPTURE(obj) obj=::kj::cp(obj)
```

Comment:
Thanks, that's a better name.

I think this is the kind of bug that we could get Microsoft to fix, but it would nevertheless mean waiting a long while before we could remove the macro. I really like the non-const-copy strategy conceptually, so I'd be sad to see it go, but new code is ultimately going to trip over this bug repeatedly.

My biggest concern would be whether or not it affects user code. I.e., how many support requests are going to be generated on the mailing list by confused users trying to port their code to Windows? If that number is low, then I'd say let's not worry about it. If it's high, then sacrificing transitive const safety might be the lesser of two evils.